project1-list/app/src/main/java/net/serkanbal/project1/TodoitemActivity.java

```diff
@@ -0,0 +1,243 @@
+            public boolean onItemLongClick(AdapterView<?> parent, View view, final int position, long id) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(TodoitemActivity.this);
+                builder.setTitle("Please Select")
+                        .setMessage("What do you want to do with the list item: " + mItemList.get(position).getItemTitle()+"?")
+
+                        .setPositiveButton("Remove", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialogInterface, int i) {
+                                mItemList.remove(position);
+                                mItemAdapter.notifyDataSetChanged();
+                            }
+                        })
+
+                        .setNeutralButton("Rename", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                LayoutInflater layoutInflater = LayoutInflater.from(TodoitemActivity.this);
+                                final View promptsView = layoutInflater.inflate(R.layout.dialogitem_layout, null);
+                                AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(
+                                        TodoitemActivity.this);
+
+                                // set prompts.xml to alertdialog builder
+                                alertDialogBuilder.setView(promptsView);
+
+                                // set dialog message
+                                alertDialogBuilder
+                                        .setCancelable(false)
+                                        .setPositiveButton("OK",
+                                                new DialogInterface.OnClickListener() {
+                                                    public void onClick(DialogInterface dialog,int id) {
+                                                        EditText inputTitle = (EditText) promptsView.findViewById(R.id.edititemtitle);
+                                                        String inputTitleUser = inputTitle.getText().toString();
+                                                        EditText inputDesc = (EditText) promptsView.findViewById(R.id.edititemdesc);
+                                                        String inputDescUser = inputDesc.getText().toString();
+                                                        String result = "";
+                                                        if (inputDescUser.contains("\n") == true) {
+                                                            String[] split = inputDescUser.split("\n+");
+                                                            for (int i = split.length - 1; i >= 0; i--) {
+                                                                result = (i + 1) + ". " + split[i] + "\n" + result;
```

Comment:
This is very clever! Another way to accomplish this would be to use some html when you put this value into its TextView later - specifically check out the ordered list and list item html tags (`<ol><li>description text</li></ol>`)